### DISCUSSION TITLE
v1.0.0b1 First working prototype

### DISCUSSION THREAD

**Discussion by antoine2711** [OpenRefine/SparqlExtension#29] [Announcements]
This is the first beta version of the SPARQL Importer for OpenRefine.

It should be compatible with OpenRefine, v3.5 and +.

<hr /><em>This discussion was created from the release <a href='https://github.com/OpenRefine/SparqlExtension/releases/tag/v1.0-beta.1'>v1.0.0b1 First working prototype</a>.</em>

**Comment by VladimirAlexiev** [comment]
## Suggestions
* Use the returned SPARQL variable names, not `Column1, Column2...`
* Idea: enable the use of query links rather than query text.

### Use query link
Example from WD:
* the short link https://w.wiki/5ah$
* is expanded to the long link
https://query.wikidata.org/#SELECT%20%3Fitem1%20%3Fitem1Label%20%3Fitem2%20%3Fitem2Label%20%3Fmic%20%7B%0A%20%20%3Fitem1%20wdt%3AP7534%20%3Fmic.%0A%20%20%3Fitem2%20wdt%3AP7534%20%3Fmic.%0A%20%20filter%28str%28%3Fitem1%29%20%3C%20str%28%3Fitem2%29%29%0A%20%20SERVICE%20wikibase%3Alabel%20%7B%20bd%3AserviceParam%20wikibase%3Alanguage%20%22%5BAUTO_LANGUAGE%5D%2Cen%22.%20%7D%0A%7D%20order%20by%20%3Fmic
* which then results in this query
```sparql
SELECT ?item1 ?item1Label ?item2 ?item2Label ?mic {
  ?item1 wdt:P7534 ?mic.
  ?item2 wdt:P7534 ?mic.
  filter(str(?item1) < str(?item2))
  SERVICE wikibase:label { bd:serviceParam wikibase:language "[AUTO_LANGUAGE],en". }
} order by ?mic
```

Ontotext GraphDB doesn't have a builtin link shortener, but you can get a query link that has the query encoded

You'll notice these query links are to the SPARQL editor (UI).
It would also be great if you auto-fill the "SPARQL endpoint" field based on the UI link.

## Doc Suggestions 
for the google doc https://docs.google.com/document/d/12wKdWkPLNIDICY_u84650xT8xvMTYjSfuQltbsmASMU/edit#

* in  use Page Setup> Pageless. Then you can increase image size so the images are readable. (Suggestion: put each image in a centered paragraph, don't lay it out on top of text)
* Add  to **Links and documents**: "Provide feedback: https://github.com/OpenRefine/SparqlExtension/discussions/29:
* "There are two fields that need to be provided" show them as text (not just image) so people can retry them more easily
